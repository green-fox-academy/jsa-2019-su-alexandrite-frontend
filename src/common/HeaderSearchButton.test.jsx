import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { useNavigation } from 'react-navigation-hooks';
import MockDate from 'mockdate';

import SearchButton from './HeaderSearchButton';

jest.mock('react-navigation-hooks');

MockDate.set('2020-01-10T04:46:05.638Z');

describe('<SearchButton />', () => {
  it('should call navigation.push when detail button is pressed', () => {
    const push = jest.fn(() => { });
    useNavigation.mockReturnValue({ push });
    const wrapper = shallow(
      <SearchButton />,
    );
    wrapper.find('TouchableHighlight')
      .first()
      .simulate('press');
    expect(useNavigation).toHaveBeenCalled();
    expect(push).toHaveBeenCalled();
  });

  it('should render properly', () => {
    const tree = renderer.create(
      <SearchButton />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
