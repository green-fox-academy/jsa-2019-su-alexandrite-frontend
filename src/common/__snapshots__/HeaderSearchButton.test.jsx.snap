// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchButton /> should render properly 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "padding": 15,
      "paddingVertical": 10,
    }
  }
>
  <Text />
</View>
`;
